### ISSUE TITLE
rdesktop cannot find Kerberos library

### ISSUE THREAD

**Issue by direwolfie** [voidlinux/void-packages#6926]
Hi everyone,

Rdesktop cannot find the kerberos library out of the box:
`can't open /usr/lib/libgssapi_krb5.so: /usr/lib/libgssapi_krb5.so: cannot open shared object file: No such file or directory`

From the package query it can be seen the the issue is with the file name:
`xbps-query -o '*libgssapi*'`
`mit-krb5-libs-1.15.1_1: /usr/lib/libgssapi_krb5.so.2.2 (regular file)`
`mit-krb5-libs-1.15.1_1: /usr/lib/libgssapi_krb5.so.2 -> /usr/lib/libgssapi_krb5.so.2.2 (link)`

As always, big thanks for all the developers and maintainers of Void Linux! You guys are the best! :)
Regards

**Comment by ebfe** [comment]
Try installing mit-krb5-devel.

**Comment by direwolfie** [comment]
Thanks for the suggestion. But the workaround is rather simple really - sim-linking current lib to a name that is required by rdesktop. And voila everything works.

Raised this issue mostly as I think that maybe it is worth adding one of the following solutions:
**rdesktop:** looks for a new lib `/usr/lib/libgssapi_krb5.so.2` instead of old one `/usr/lib/libgssapi_krb5.so`; OR
**krb5:** creates another link for compatibility reasons `/usr/lib/libgssapi_krb5.so`.

(In advance - I know very little about package management, so please forgive my ignorance).

**Comment by ebfe** [comment]
> Thanks for the suggestion. But the workaround is rather simple really - sim-linking current lib to a name 
> that is required by rdesktop. And voila everything works.

But it's a workaround and breaks if mit-krb5 is updated. The symlink is owned by the mit-krb5-devel package.

**Comment by pullmoll** [comment]
BTW: `mit-krb5-devel` should `vmove usr/lib/pkgconfig` and IMO a patch for `rdesktop` to resolve the current major sover of `libgssapi_krb5.so` (perhaps using readlink) and use that name to dynload it later on would be a good solution.

**Comment by direwolfie** [comment]
> But it's a workaround and breaks if mit-krb5 is updated. The symlink is owned by the mit-krb5-devel package.

Was not aware of that. Thanks for pointing it out!

> IMO a patch for rdesktop to resolve the current major sover of libgssapi_krb5.so (perhaps using readlink) and use that name to dynload it later on would be a good solution.

Was wondering, does this solution means that `mit-krb5-devel` will not be required and the libs are enough?

**Comment by pullmoll** [comment]
> Was wondering, does this solution means that mit-krb5-devel will not be required and the libs are enough?

Yes. It's for how Void generally moves `/usr/lib/*.so` files to the `<pkgname>-devel` subpackages. Of course it could be argued that Void's design decision to do that is wrong. If it didn't move the *.so files, the problem you described would not exist.

**Comment by direwolfie** [comment]
Thanks for explaining that!
